src/ralph/networks/models/networks.py

```diff
@@ -0,0 +1,414 @@
+        null=True,
+        blank=True,
+        default=None,
+    )
+    racks = models.ManyToManyField(
+        'data_center.Rack',
+        verbose_name=_('racks'),
+        blank=True,
+    )
+    network_environment = models.ForeignKey(
+        NetworkEnvironment,
+        verbose_name=_('environment'),
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+    )
+    min_ip = models.DecimalField(
+        verbose_name=_('smallest IP number'),
+        editable=False,
+        max_digits=39,
+        decimal_places=0,
+    )
+    max_ip = models.DecimalField(
+        verbose_name=_('largest IP number'),
+        editable=False,
+        max_digits=39,
+        decimal_places=0,
+    )
+    kind = models.ForeignKey(
+        NetworkKind,
+        verbose_name=_('network kind'),
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    dhcp_broadcast = models.BooleanField(
+        verbose_name=_('Broadcast in DHCP configuration'),
+        default=True,
+        db_index=True,
+    )
+
+    @cached_property
+    def network(self):
+        return ipaddress.ip_network(self.address, strict=False)
+
+    @property
+    def network_address(self):
+        return self.network.network_address
+
+    @property
+    def broadcast_address(self):
+        return self.network.broadcast_address
+
+    @property
+    def netmask(self):
+        return self.network.prefixlen
+
+    @property
+    def gateway(self):
+        ip = self.ips.filter(is_gateway=True).first()
+        return ip.ip if ip else None
+
+    @property
+    def data_center(self):
+        if not self.network_environment:
+            return None
+        return self.network_environment.data_center
+
+    @property
+    def size(self):
+        # TODO: IPv6
+        if not self.min_ip and not self.max_ip or self.netmask == 32:
+            return 0
+        if self.netmask == 31:
+            return 2
+        return self.max_ip - self.min_ip - 1
+
+    class Meta:
+        verbose_name = _('network')
+        verbose_name_plural = _('networks')
+        unique_together = ('min_ip', 'max_ip')
+
+    def __str__(self):
+        return '{} ({})'.format(self.name, self.address)
+
+    def save(self, *args, **kwargs):
+        """
+        Override standard save method. Method saves min_ip and max_ip
+        depenfing on the self.address.
+
+        Example:
+            >>> network = Network.objects.create(
+            ...     name='C', address='192.168.1.0/24'
+            ... )
+            >>> network.min_ip, network.max_ip, network.gateway
+            (3232235776, 3232236031, None)
+        """
+        self.min_ip = int(self.network_address)
+        self.max_ip = int(self.broadcast_address)
+        self._assign_parent()
+        super(Network, self).save(*args, **kwargs)
+        self._assign_ips_to_network()
+
+    def _assign_ips_to_network(self):
+        IPAddress.objects.exclude(
+            network=self
```

Comment:
add condition for descendants here, sth like `exclude(network__in=self.get_descendants())` - we don't want to reassign all IPs in big network (ex. from subnetworks)